tvx/suite_messages_paych.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+
+	abi_spec "github.com/filecoin-project/specs-actors/actors/abi"
+	big_spec "github.com/filecoin-project/specs-actors/actors/abi/big"
+	paych_spec "github.com/filecoin-project/specs-actors/actors/builtin/paych"
+	crypto_spec "github.com/filecoin-project/specs-actors/actors/crypto"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/oni/tvx/chain"
+	"github.com/filecoin-project/oni/tvx/drivers"
+)
+
+func MessageTest_Paych() error {
+	var initialBal = abi_spec.NewTokenAmount(200_000_000_000)
+	var toSend = abi_spec.NewTokenAmount(10_000)
+
+	err := func(testname string) error {
+		td := drivers.NewTestDriver()
+
+		v := newEmptyMessageVector()
+
+		// will create and send on payment channel
+		sender, senderID := td.NewAccountActor(drivers.SECP, initialBal)
+
+		// will be receiver on paych
+		receiver, receiverID := td.NewAccountActor(drivers.SECP, initialBal)
+
+		preroot := td.GetStateRoot()
+
+		// the _expected_ address of the payment channel
+		paychAddr := chain.MustNewIDAddr(chain.MustIdFromAddress(receiverID) + 1)
+		createRet := td.ComputeInitActorExecReturn(sender, 0, 0, paychAddr)
+
+		msg := td.MessageProducer.CreatePaymentChannelActor(sender, receiver, chain.Value(toSend), chain.Nonce(0))
+		v.ApplyMessages = append(v.ApplyMessages, Message{Bytes: chain.MustSerialize(msg)})
+
+		// init actor creates the payment channel
+		td.ApplyExpect(
+			msg,
+			chain.MustSerialize(&createRet))
+
+		var pcState paych_spec.State
+		td.GetActorState(paychAddr, &pcState)
+		assert.Equal(t, senderID, pcState.From)
+		assert.Equal(t, receiverID, pcState.To)
+		td.AssertBalance(paychAddr, toSend)
+
+		postroot := td.GetStateRoot()
+
+		v.CAR = td.MustMarshalGzippedCAR(preroot, postroot)
+		v.Pre.StateTree.RootCID = preroot
+		v.Post.StateTree.RootCID = postroot
+
+		// encode and output
+		enc := json.NewEncoder(os.Stdout)
+		if err := enc.Encode(&v); err != nil {
+			return err
+		}
+
+		return nil
+	}("happy path constructor")
+	if err != nil {
+		return err
+	}
+
+	err = func(testname string) error {
+		td := drivers.NewTestDriver()
+
+		v := newEmptyMessageVector()
+
+		//const pcTimeLock = abi_spec.ChainEpoch(1)
+		const pcTimeLock = abi_spec.ChainEpoch(0)
```

Comment:
This is a difference between our tests and `chain-validation` - it seems like they are using epoch 1 as an initial epoch for tests, whereas we are using 0.